### PR TITLE
Created library

### PR DESCRIPTION
**Pull request by myslimov**
Did simple customizable toggle switch.

### CODE REVIEW

File: projects/switcher/src/lib/switcher.service.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { SwitcherI } from "./models/config.model";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SwitcherService {
+
+  constructor(@Inject(CONFIG_TOKEN) private styleConfig: SwitcherI) { }
+
+  getStyles(): any {
```
Comment (a-polikarpov) [inline projects/switcher/src/lib/switcher.service.ts:12]: @myslimov why `any`?

File: projects/switcher/src/lib/switcher/switcher.component.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { Component, forwardRef, Inject, Input, OnInit } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from "@angular/forms";
+import { CONFIG_TOKEN } from "../switcher.module";
+import { SwitcherI } from "../models/config.model";
+import { SwitcherService } from "../switcher.service";
+
+@Component({
```
Comment (a-polikarpov) [inline projects/switcher/src/lib/switcher/switcher.component.ts:7]: @myslimov use `onPush` strategy

File: projects/switcher/src/lib/switcher/switcher.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  disable: boolean = false;
+  @Input('bgColor') bg: string = ''
+  onChange = (value: boolean) => {};
+  onTouch = () => {};
+
+  constructor(private styleService: SwitcherService) {
+    const {sliderChecked, sliderDefault, backgroundColor} = this.styleService.getStyles()
+    this.bgCheckedValue = sliderChecked;
+    this.bgDefaultValue = sliderDefault;
+    this.bg = backgroundColor;
```
Comment (a-polikarpov) [inline projects/switcher/src/lib/switcher/switcher.component.ts:32]: @myslimov init it where you define it. Why do you duplicate this all?

File: projects/switcher/src/public-api.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Public API Surface of switcher
+ */
+
+export * from './lib/switcher.service';
+export * from './lib/switcher/switcher.component';
+export * from './lib/switcher.module'
+
+export const a = 4;
```
Comment (a-polikarpov) [inline projects/switcher/src/public-api.ts:9]: @myslimov why is it here?

**Review by a-polikarpov — CHANGES_REQUESTED** [review]

File: projects/switcher/src/public-api.ts (hunk 4 above)
Comment (myslimov) [reply projects/switcher/src/public-api.ts:9]: @a-polikarpov The delicate balance of the universe depends on this variable :)

File: projects/switcher/src/lib/switcher.service.ts [hunk 5]
```diff
@@ -9,8 +9,12 @@ export class SwitcherService {
...
 
-  getStyles(): any {
-    return this.styleConfig;
+  public get(property: string): string {
+    return {
+      defaultSlider: this.styleConfig.sliderDefault,
+      checkedSlider: this.styleConfig.sliderChecked,
+      background: this.styleConfig.backgroundColor
+    }[property] || ''
   }
```
Comment (myslimov) [inline projects/switcher/src/lib/switcher.service.ts:19]: extra emty line, just saw it, fix it after your review

File: projects/switcher/src/lib/switcher/switcher.component.html [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    [ngClass]="disable ? 'disabled' : '' "
+    [ngStyle]="{'background-color': bg}"
+  >
+    <input
+      id="switcher"
+      type="checkbox"
+      class="toggle-button">
+    <label
+      (click)="changeCheck()"
+      [ngStyle]="checked ? {'background-color': checkedSlider} : {'background-color': defaultSlider}"
```
Comment (a-polikarpov) [inline projects/switcher/src/lib/switcher/switcher.component.html:13]: @myslimov move it to getter

File: projects/switcher/src/public-api.ts (hunk 4 above)
Comment (a-polikarpov) [reply projects/switcher/src/public-api.ts:9]: remove it

File: projects/switcher/src/lib/switcher/switcher.component.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from "@angular/forms";
+import { SwitcherService } from "../switcher.service";
+
+@Component({
+  selector: 'switcher',
+  templateUrl: './switcher.component.html',
+  styleUrls: ['./switcher.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (a-polikarpov) [inline projects/switcher/src/lib/switcher/switcher.component.ts:9]: @myslimov you don't use markForCheck() in the component. Did you test the component properly, that it works correctly in all cases?

**Review by a-polikarpov — CHANGES_REQUESTED** [review]

**Pull request merged**